src/classes/PS_IntegrationService_TEST.cls

```diff
@@ -35,16 +35,112 @@
+    @IsTest
+    private static void shouldSetConfigurationWithInvalidConfigurationParameters() {
+       configParams = new Map<String, List<Map<String, String>>>();
+
+        Test.startTest();
+        try {
+            PS_IntegrationService.setConfiguration(configParams);
+        } catch (PS_IntegrationService.IntegrationServiceException ex) {
+            System.assertEquals(PS_IntegrationService.INVALID_CONFIG_PARAMS, ex.getMessage());
+        }
+        Test.stopTest();
+    }
+
+    /*******************************************************************************************************
+   * @description Helper method to build test configuration parameters
+   */
+    private static void buildConfigurationParams() {
+     List<Map<String, String>> applicationConfig = new List<Map<String, String>>();
+     configParams = new Map<String, List<Map<String, String>>>();
+     paymentConfigParams = new Map<String, PS_IntegrationService.Config> {
+             'iss' => new PS_IntegrationService.Config('iss', 'ApiKeyManager'),
+             'sub' => new PS_IntegrationService.Config('sub', 'voduyemi'),
+             'sfdo.roles' => new PS_IntegrationService.Config('sfdo.roles',
+                     'Payments.Purchase, Payments.Refund, Payments.Commitment'),
+             'sfdo.product' => new PS_IntegrationService.Config('sfdo.product', 'CRM'),
+             'apiKey' => new PS_IntegrationService.Config('apiKey', 'a5146b4b-3d2b-4fc4-a5e5-8387968859a4'),
+             'sfdo.merchantids' => new PS_IntegrationService.Config('sfdo.merchantids',
+                     '47b70ca4-0f6e-4c52-819f-667ae804f9d2'),
+             'gatewayids' => new PS_IntegrationService.Config('gatewayids',
+                     'f23da98a-c7aa-4637-962a-8c852e765e38'),
+             'baseURL' => new PS_IntegrationService.Config('baseURL',
+                     'https://SOMEURL.execute-api.us-west-2.amazonaws.com/v1')
+     };
+
+     makanaConfigParams = new Map<String, PS_IntegrationService.Config> {
+             'apiKey' => new PS_IntegrationService.Config('apiKey',
+                     'a5146b4b-3d2b-4fc4-a5e5-8387968859a4-makana'),
+             'baseURL' => new PS_IntegrationService.Config('baseURL',
+                     'https://SOMEURL.execute-api.us-west-2.amazonaws.com/v1')
+     };
+
+     for(String field : paymentConfigParams.keySet()) {
+      Map<String, String> stringMap = new Map<String, String>();
+      stringMap.put(PS_IntegrationService.KEY_CONST, field);
+      stringMap.put(PS_IntegrationService.VALUE_CONST, paymentConfigParams.get(field).value);
+      stringMap.put(PS_IntegrationService.IS_SECRET_CONST, field == API_KEY ? 'true' : 'false');
+      applicationConfig.add(stringMap);
+     }
+
+     configParams.put(PAYMENTS_APPLICATION_TYPE, applicationConfig);
+     applicationConfig = new List<Map<String, String>>();
+
+     for (String field : makanaConfigParams.keySet()) {
+      Map<String, String> stringMap = new Map<String, String>();
```

Comment:
More mis-aligned indents - should be four spaces for this whole block from the for loop